roofit/roofit/src/RooPolyFunc.cxx

```diff
@@ -0,0 +1,289 @@
+/*****************************************************************************
+ * Project: RooFit                                                           *
+ * Package: RooFitModels                                                     *
+ * @(#)root/roofit:$Id$
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2005, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+/** \class RooPolyFunc
+    \ingroup Roofit
+
+RooPolyFunc implements a polynomial function in multi-variables.
```

Comment:
```suggestion
RooPolyFunc implements a polynomial function in multi-variables.
```

Can you explain this class a bit more in the documentation? You should probably warn the user that different from RooPolynomial, the polynomial coefficients are not part of the RooFit computation graph! Because that's why your unit test fails: you were changing the values of RooRealVars for the coefficients `after` creating the `RooPolyFunc`, and wrongly assumed that the changed coefficient values will propagate.